tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java

```diff
@@ -0,0 +1,218 @@
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+@Feature(PROFILING)
+@Story(DEFAULT_CORE_EVENT_TRACER)
+public class ExportConfigurationChangeTestCase extends
+    MuleArtifactFunctionalTestCase implements OpenTelemetryTracingTestRunnerConfigAnnotation {
+
+  private static final String EXPECTED_FLOW_SPAN_NAME = "mule:flow";
+  private static final String EXPECTED_SET_PAYLOAD_SPAN_NAME = "mule:set-payload";
+  private static final String FLOW_LOCATION = "flow";
+
+  private static final String TEST_ARTIFACT_ID = "ExportConfigurationChangeTestCase#test";
+
+  public static final int TIMEOUT_MILLIS = 30000;
+
+  private static final int POLL_DELAY_MILLIS = 100;
+  public static final int MAX_BACKOFF_ATTEMPTS = 2;
+
+  private static final String SET_PAYLOAD_LOCATION = "flow/processors/0";
+  private Path configFile;
+  private File file;
+
+  @Override
+  protected String getConfigFile() {
+    return "tracing/export-configuration-change.xml";
+  }
+
+  @ClassRule
+  public static final TestServerRule originalServer = new TestServerRule();
+
+  @ClassRule
+  public static final TestServerRule afterConfigurationChangeServer = new TestServerRule();
+
+  @Override
+  protected void doSetUpBeforeMuleContextCreation() throws Exception {
+    file = createTempFile("tracing", "test");
+    configFile = Paths.get(getClass().getResource("/tracing/config/exporter.conf").getPath());
+    copy(configFile, Paths.get(file.getPath()), REPLACE_EXISTING);
+    setProperty(MULE_OPEN_TELEMETRY_TRACING_CONFIGURATION_FILE_PATH, file.getAbsolutePath());
+    setProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENABLED, TRUE.toString());
+    setProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENDPOINT,
+                "http://localhost:" + originalServer.httpPort());
+    setProperty(MULE_OPEN_TELEMETRY_EXPORTER_CONFIGURATION_WATCHER_DEFAULT_DELAY_PROPERTY, "1000");
+  }
+
+  @After
+  public void after() {
+    clearProperty(MULE_OPEN_TELEMETRY_TRACING_CONFIGURATION_FILE_PATH);
+    clearProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENDPOINT);
+    clearProperty(MULE_OPEN_TELEMETRY_EXPORTER_CONFIGURATION_WATCHER_DEFAULT_DELAY_PROPERTY);
+    clearProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENABLED);
+  }
+
+  @Test
+  public void test() throws Exception {
+    flowRunner(FLOW_LOCATION).withPayload(TEST_PAYLOAD).run().getMessage();
+
+    pollTillExportedSpansCaptured(originalServer);
+
+    List<String> attributesToAssertExistence = getDefaultAttributesToAssertExistence();
+
+    Collection<CapturedExportedSpan> exportedSpans = originalServer.getCapturedExportedSpans();
+
+    Map<String, String> setPayloadAttributeMap = createAttributeMap(SET_PAYLOAD_LOCATION, TEST_ARTIFACT_ID);
+
+    assertExpectedSpanTree(attributesToAssertExistence, exportedSpans, setPayloadAttributeMap);
+
+    // We update the file by recopying it.
+    setProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENDPOINT,
+                "http://localhost:" + afterConfigurationChangeServer.httpPort());
```

Comment:
Does the httpPort change from the original one? Isn't it the same?

Reply:
Changed to make sure that this works in all testing envs.

Reply:
Yes, the port to which export also changes.

Reply:
Basically there are two servers that are created for the export of the telemetry. After the change in the file it begins to export the other server (collector).